### PR TITLE
SC: remove uri in requetValueTransferEvent and add the routine to get uri

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes
This PR removed uri in requetValueTransferEvent and add the routine to get uri before handling value transfer.

This PR can reduce the gas used of request value transfer transactions.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: node/sc/bridge_manager.go [hunk 1]
```diff
@@ -311,7 +314,19 @@ func (bi *BridgeInfo) handleRequestValueTransferEvent(ev *RequestValueTransferEv
 		}
 		logger.Trace("Bridge succeeded to HandleERC20Transfer", "nonce", ev.RequestNonce, "tx", handleTx.Hash().String())
 	case ERC721:
-		handleTx, err = bi.bridge.HandleERC721Transfer(auth, ev.Raw.TxHash, ev.From, ev.To, tokenAddr, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, ev.Uri, ev.ExtraData)
+		// get URI of the ERC721
+		var uri string
+		erc721, err := scnft.NewERC721Metadata(ev.TokenAddress, bi.counterpartBackend)
```
Comment (jeonghonah) [inline node/sc/bridge_manager.go]: Is it safe to call every time to handle events?

Comment (KimKyungup) [reply node/sc/bridge_manager.go]: I think it's safe. Because after minting NFT, owner can't change it in normal ERC721Metadata.

Comment (jeonghonah) [reply node/sc/bridge_manager.go]: How about caching the erc721 object not to recreate it for each event?

Comment (KimKyungup) [reply node/sc/bridge_manager.go]: I think there is not big processing during create ERC721 object. 
But, anyway can I consider it in next PR.

File: node/sc/bridge_manager.go [hunk 2]
```diff
@@ -311,7 +314,19 @@ func (bi *BridgeInfo) handleRequestValueTransferEvent(ev *RequestValueTransferEv
...
+		if err != nil {
+			logger.Warn("Failed to get erc721 token instance", "erc721", ev.TokenAddress.String(), "onParent", bi.onChildChain)
+		} else {
+			uri, err = erc721.TokenURI(nil, ev.ValueOrTokenId)
+			if err != nil {
+				logger.Warn("Failed to get URI of the ERC721 token", "erc721", ev.TokenAddress.String(), "onParent", bi.onChildChain, "tokenId", ev.ValueOrTokenId.String())
+			}
+		}
+
+		handleTx, err = bi.bridge.HandleERC721Transfer(auth, ev.Raw.TxHash, ev.From, ev.To, tokenAddr, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, uri, ev.ExtraData)
```
Comment (nogang) [inline node/sc/bridge_manager.go]: Does this matter if the err is not nil?

Comment (KimKyungup) [reply node/sc/bridge_manager.go]: Yes. If this handling fails, bridge node should retry to handle it again. So this code return the error.

Comment (nogang) [reply node/sc/bridge_manager.go]: Sorry. The comment is wrong. "323: uri, err = erc721.TokenURI (nil, ev.ValueOrTokenId)" was asking if you can use uri when err is not nil.

Comment (KimKyungup) [reply node/sc/bridge_manager.go]: I thought if the erc721 doen't support URI (metadata), skipping is ok.
But this code can skip other error cases like communication error. So I will try to check specific `revert` error. :) Thanks, pal.

**Comment by KimKyungup** [comment]
This PR needed https://github.com/klaytn/klaytn/pull/225.
I will rebase this after merging it.

File: node/sc/bridge_manager.go (hunk 2 above)
Comment (KimKyungup) [reply node/sc/bridge_manager.go]: I rebased it. :)

**Comment by KimKyungup** [comment]
@nogang I rebased this for resolving a conflict.

File: node/sc/bridge_manager.go [hunk 3]
```diff
@@ -311,7 +315,23 @@ func (bi *BridgeInfo) handleRequestValueTransferEvent(ev *RequestValueTransferEv
 		}
 		logger.Trace("Bridge succeeded to HandleERC20Transfer", "nonce", ev.RequestNonce, "tx", handleTx.Hash().String())
 	case ERC721:
-		handleTx, err = bi.bridge.HandleERC721Transfer(auth, ev.Raw.TxHash, ev.From, ev.To, tokenAddr, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, ev.Uri, ev.ExtraData)
+		// get URI of the ERC721
+		var uri string
+		erc721, err := scnft.NewERC721Metadata(ev.TokenAddress, bi.counterpartBackend)
+		if err != nil {
+			logger.Warn("Failed to get erc721 token instance", "erc721", ev.TokenAddress.String(), "onParent", bi.onChildChain)
```
Comment (laggu) [inline node/sc/bridge_manager.go]: if err is not nil, the `uri` has empty string.
is it okay not to return here??
`uri` will be used on line 334

Comment (KimKyungup) [reply node/sc/bridge_manager.go]: Thanks. @laggu :)
I fixed the code as you commented.

**Comment by KimKyungup** [comment]
@nogang NoGang? :)

**Pull request merged**